### ISSUE TITLE
Add AI Optimizer

### ISSUE THREAD

**Issue by ninja7v** [ninja7v/Traffic-Simulator#3]
local/global?
-gradient descent
-neural network
-rules
-update car itinerary
